Add documentation for accessibility features (#254)
eclipse-glsp/glsp-client: packages/client/src/features/accessibility/README.md

# Keyboard-Only: Accessibility Features

The features presented in the following are introduced for disability-aware conceptual modeling and allow basic user model interactions to be performed only with the keyboard.

## Getting started

In order to activate and use all of the currently provided keyboard only accessibility features, the module `glspAccessibilityModule` needs to be in use.
To activate and use only specific accessibility features, the respective module (see list below) needs to be in use.

## Features

Below is a list of the keyboard only accessibility features:<br>

| Feature |                                           Description                                            |             Shortcut              |          Module           |
| ------- | :----------------------------------------------------------------------------------------------: | :-------------------------------: | :-----------------------: |

> | Search  | Search for elements or edges using labels, types, etc. <br> with respective diagram highlighting |             `CTRL+F`              | `glspSearchPaletteModule` |

Reviewer: Just a suggestion: You could use for the shortcuts in this document also the keybinding tag, e.g. `<kbd>CTRL</kbd>+<kbd>F</kbd>` to get <kbd>CTRL</kbd>+<kbd>F</kbd>

| Move    |                  Move elements, edges or viewport into the directions ← ↑ → ↓.                   |             `← ↑ → ↓`             |   `glspMoveZoomModule`    |

> | Zoom    |                        Zoom inor out gradually to elemenets or viewport.                         |       `+` & `-` & `CTRL+0`        |   `glspMoveZoomModule`    |

Reviewer: ```suggestion
| Zoom    |                        Zoom in or out gradually to elemenets or viewport.                         |       `+` || `-` || `CTRL+0`        |   `glspMoveZoomModule`    |
```

> | Resize  |                          Increase or descrease element sizes gradually.                          | `ALT+A` then `+` & `-` & `CTRL+0` |   `glspResizeKeyModule`   |

Reviewer: ```suggestion
| Resize  |                          Increase or descrease element sizes gradually.                          | `ALT+A` then `+` || `-` || `CTRL+0` |   `glspResizeKeyModule`   |
```

## Usage

#### Search

The search palette allows to search labelled elements or edges that have a labelled node as source or target. The result set will be highlighted accordingly, to also be able to visually distinguish between the searched and the remaining parts of the given diagram.

#### Read / Select via Search

1. `CTRL + F`: Open the search palette

> 2. Type in either **type** or **label** of element/edge

Reviewer: ```suggestion
2. In the palette's input field enter either the **type** or the **label** of nodes or edges. The arrow keys can be used to navigate in the search result list.
```

> 3. `ENTER`: Select the element/edge

Reviewer: ```suggestion
3. `ENTER`: Select the node or edge that is highlighted in the search result list.
```

### Model Exploration

#### Move

The move functionality can be used to move parts of the model, such as selected nodes or edges or the whole viewport in all directions.

##### Steps

1. Select one or multiple elements or the viewport.
2. Use arrow keys (← ↑ → ↓) to move element(s) or viewport to the desired location.

#### Zoom

This zoom functionality can be used to gradually adapt the zoom level of one element, a set of elements or the
viewport.

#### Steps

1. Select one or multiple elements or the viewport.
2. Use `+` or `-` to zoom in or out gradually.
3. `CTRL+0`: Set the zoom level to default.

### Resize element

> The resize functionality helps to set the size of the nodes, by either increasing or decreasin the size of the nodes’ shape gradually.

Reviewer: ```suggestion
The resize functionality helps to set the size of the nodes, by either increasing or decreasing the size of the nodes’ shape gradually.
```

> #### Steps

Reviewer: Please also mention how to deactivate the resize mode.
